docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -0,0 +1,85 @@
+# ADR 054: Crypto encoding (part 2)
+
+## Changelog
+
+\*2020-2-27: Created
+
+## Context
+
+Amino has been a pain point of many users in the ecosystem. While Tendermint does not suffer greatly from the performance degradation introduced by amino, we are making an effort in moving the encoding format to a widely adopted format, [Protocol Buffers](https://developers.google.com/protocol-buffers). With this migration a new standard is needed for the encoding of keys. This will cause ecosystem wide breaking changes.
+
+Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
+
+> With keys becoming a [oneof type](https://developers.google.com/protocol-buffers/docs/proto3#oneof) we have the option of removing the prefix bytes.
```

Comment:
What _are_ the prefix bytes?

Reply:
removed this comment as it was relevant before when I was debating on keeping the prefix bytes, but that was based on speaking with people about encoded keys. Looking through the code, this didn't seem like the case.